robotoff/app/core.py

```diff
@@ -67,6 +71,23 @@ def get_insights(
 
     query = ProductInsight.select()
 
+    def join_with_no_votes_by(query, criteria):
+        return query.join(
+            AnnotationVote,
+            join_type=peewee.JOIN.LEFT_OUTER,
+            on=((AnnotationVote.insight_id == ProductInsight.id) & (criteria)),
+        ).where(AnnotationVote.id.is_null())
+
+    if avoid_voted_by_username:
+        query = join_with_no_votes_by(
+            query, (AnnotationVote.username == avoid_voted_by_username)
+        )
+
+    if avoid_voted_by_device_id:
+        query = join_with_no_votes_by(
+            query, (AnnotationVote.device_id == avoid_voted_by_device_id)
+        )
```

Comment:
It would actually work with both, but it would be a potentially more expensive JOIN. We could of course enforce that it's not used that way, but it didn't feel important. WDYT?

Reply:
@kulizhsy does that mean you volunteer to implement it? :)

Reply:
Sure, I'll do it :)

Reply:
And I see it's done. Nice!